### ISSUE TITLE
Add `recreateClosed` automatically if PR title/groupName isn't unique

### ISSUE THREAD

**Issue by Shegox** [renovatebot/renovate#18183]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.222.2

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.5.5

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

If you use a `groupName`, which doesn't include uniqueness (e.g. not a version number or specific digest) Renovate will treat the first merged PR as closed and won't open subsequent PRs. 

To reproduce it use a group like this: 
```json
{
    "packageRules": [
        {
            "matchDatasources": [
                "docker"
            ],
            "matchUpdateTypes": [
                "digest"
            ],
            "groupName": "docker image digests"
        }
    ]
}
```

I'm not sure if it is possible to auto-detect this scenario and set `"recreateClosed": true` for these kind of PRs to make them immortal and receive digest updates for grouped PRs. 
Alternatively we could add to the documentation that you should set `"recreateClosed": true` if your `groupName` doesn't include a variable. 

### Related Issues 
- #17603
- #11772
- Slack conversation: https://renovatebot.slack.com/archives/CAFH752JU/p1665128489447739

### Relevant debug logs

<details><summary>Logs</summary>

```
2022-10-06T07:39:21.398958273Z DEBUG: getPrList success (repository=some-org/some-repo)
2022-10-06T07:39:21.399012078Z        "pullsTotal": 254,
2022-10-06T07:39:21.399750284Z        "requestsTotal": 1,
2022-10-06T07:39:21.399764753Z        "apiQuotaAffected": true
...

2022-10-06T07:39:21.441122708Z DEBUG: getBranchPr(renovate/docker-image-digests) (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.441151790Z DEBUG: findPr(renovate/docker-image-digests, undefined, open) (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.441347064Z DEBUG: findPr(renovate/docker-image-digests, undefined, closed) (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.441417122Z DEBUG: branchExists=false (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.441475139Z DEBUG: dependencyDashboardCheck=undefined (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.446653359Z DEBUG: recreateClosed is false (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.446774656Z DEBUG: findPr(renovate/docker-image-digests, chore(deps): update docker image digests, !open) (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.446875246Z DEBUG: Found PR #2050 (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.446950085Z DEBUG: Found closed PR with current title (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.447134458Z DEBUG: Returning PR from cache (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.447309675Z DEBUG: Closed PR already exists. Skipping branch. (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.447313801Z        "prTitle": "chore(deps): update docker image digests"
2022-10-06T07:39:21.447584146Z DEBUG: Merged PR is blocking this branch (repository=some-org/some-repo, branch=renovate/docker-image-digests)
2022-10-06T07:39:21.447587805Z        "pr": 2050
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by dominykas** [comment]
OK, actually, I was able to add a repro case - see https://github.com/dominykas/renovate-18183/issues/4: "These are blocked by an existing closed PR"

I merged https://github.com/dominykas/renovate-18183/pull/5, then reverted it https://github.com/dominykas/renovate-18183/pull/6 - I would expect a new PR to be opened up without any interaction.

Note how the only change that is pending is the SHA updates - I think the PR would get properly opened up if there was an actual new version pending, not just the digest.

**Comment by rarkins** [comment]
Here is the log from the app:

```
{"level":20,"branch":"renovate/docker-images","msg":"syncBranchState()","time":"2022-10-18T05:13:48.613Z"}
{"level":20,"branch":"renovate/docker-images","msg":"branch.isUpToDate(): using cached result \"true\"","time":"2022-10-18T05:13:48.619Z"}
{"level":20,"branch":"renovate/docker-images","msg":"getBranchPr(renovate/docker-images)","time":"2022-10-18T05:13:48.619Z"}
{"level":20,"branch":"renovate/docker-images","msg":"findPr(renovate/docker-images, undefined, open)","time":"2022-10-18T05:13:48.619Z"}
{"level":20,"branch":"renovate/docker-images","msg":"findPr(renovate/docker-images, undefined, closed)","time":"2022-10-18T05:13:48.619Z"}
{"level":20,"branch":"renovate/docker-images","msg":"branchExists=false","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","msg":"dependencyDashboardCheck=undefined","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","msg":"recreateClosed is false","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","msg":"findPr(renovate/docker-images, Update docker-images, !open)","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","msg":"Found PR #5","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","msg":"Found closed PR with current title","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","msg":"Returning PR from cache","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","prTitle":"Update docker-images","msg":"Closed PR already exists. Skipping branch.","time":"2022-10-18T05:13:48.620Z"}
{"level":20,"branch":"renovate/docker-images","pr":5,"msg":"Merged PR is blocking this branch","time":"2022-10-18T05:13:48.621Z"}
```

**Comment by rarkins** [comment]
<img width="460" alt="image" src="https://user-images.githubusercontent.com/6311784/196343792-2572cfba-ebc8-417d-b701-568010db20f7.png">

<img width="492" alt="image" src="https://user-images.githubusercontent.com/6311784/196343738-1fb836c5-010e-46ba-9cc7-c76cf031fe6a.png">

**Comment by rarkins** [comment]
Reproduction forked and confirmed: https://github.com/renovate-reproductions/18183/issues/2

**Comment by rarkins** [comment]
For this to work properly, one of these `if` blocks is meant to be `true`:

https://github.com/renovatebot/renovate/blob/1f70da4aede2e7e0b31fa7479ff5ea82fab572c0/lib/workers/repository/updates/generate.ts#L172-L185

**Comment by dominykas** [comment]
Thanks!
